### PR TITLE
Add a Show Command to show user details

### LINKED ISSUE #49: Add a show command
`show` command should be able to show details of the contact given their name or index in the list.

### PR DESCRIPTION
**Pull request by crypto-code**
This resolves #49 by adding the show command with the following formats:

## **Format 1**: `show #index`
If the argument is a number it is taken to be an index, and details of contact at index is shown

## **Format 2**: `show name`
If the argument is a string then it is taken to be a name, and details of contact with that name is shown
If there is more than one user with that name then it display list of the users with that name

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ShowCommand.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+     *
+     * @param index of the Person to show
+     */
+    public ShowCommand(Index index) {
+        this.predicate = null;
+        this.index = index;
+    }
+
+    @Override
+    public CommandResult execute(Model model) {
```
Comment (moreTriangles) [inline src/main/java/seedu/address/logic/commands/ShowCommand.java:61]: Nesting is quite deep, and method is very long, making it not very readable. Perhaps some parts could be abstracted out to different functions?

File: src/main/java/seedu/address/logic/commands/ShowCommand.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                return new CommandResult(String.format(MESSAGE_SUCCESS,
+                        model.getFilteredPersonList().get(index).getName()));
+            } else {
+                model.updateFilteredPersonList(predicate);
+                return new CommandResult(String.format(MESSAGE_MULTIPLE_NAME, model.getFilteredPersonList().size()));
+            }
+        } else if (index != null) {
+            int index = this.index.getZeroBased();
+            if (index >= model.getFilteredPersonList().size()) {
+                return new CommandResult(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
```
Comment (moreTriangles) [inline src/main/java/seedu/address/logic/commands/ShowCommand.java:88]: I believe alternate error paths should be thrown as `new CommandException(errror message)`, to be handled in MainWindow.java. Maybe you could refer to the execute implementation in `EditCommand.java`

**Review by moreTriangles — CHANGES_REQUESTED** [review]
Implementation is good, and tests are done 👍 . But there are some issues about code quality and error handling.

File: src/main/java/seedu/address/logic/commands/ShowCommand.java (hunk 2 above)
Comment (crypto-code) [reply src/main/java/seedu/address/logic/commands/ShowCommand.java]: Threw CommandException for invalid index and invalid name.

**Pull request merged**